### DISCUSSION TITLE
website with php

### DISCUSSION THREAD

**Discussion by strawbs1989** [community/community#124102] [Programming Help]
### Select Topic Area

General

### Body

hi all i've recently uploaded my site with a html and php folder 
for a project on my website, but everytime i refresh the site the php file keeps wanting to download?

now i personally think its something to do with this 

`<?php
// Read requests from the flat file
$file = 'requests/requests.txt';
$requests = file_get_contents($file);

// Display requests
echo nl2br($requests); // Convert newlines to <br> tags for HTML display
?>`

i embed that code on my site but keeps wanting to download? its wierd

**Comment by davevad93** [comment]
HI @strawbs1989 ,

Try adding this header to set the MIME type for the `PHP` response, this should fix it:

```php
header('Content-Type: text/html');
```

**Reply by strawbs1989** [reply]
i've added it `<?php
header('Content-Type: text/html');
// Read requests from the flat file
$file = 'requests/requests.txt';
$requests = file_get_contents($file);

// Display requests
echo nl2br($requests); // Convert newlines to <br> tags for HTML display
?>`

but didn't make no difference i've even added 
`<iframe src="requests/display_requests.php"></iframe>`
into my index.html and still downloading

**Comment by kcraftcode** [comment]
Apache Configuration:
Ensure you have a line like this in your Apache configuration:
AddType application/x-httpd-php .php
Also, ensure that the PHP module is loaded:
LoadModule php_module modules/libphp.so
Nginx Configuration:
Ensure that your server block includes something like:
location ~ \.php$ {
    include snippets/fastcgi-php.conf;
    fastcgi_pass unix:/var/run/php/php7.4-fpm.sock;
}
Check for PHP Installation: Verify that PHP is installed on your server. You can do this by running php -v in your terminal. If PHP is not installed, you'll need to install it. For example, on Ubuntu, you can install PHP with:
sudo apt update
sudo apt install php libapache2-mod-php

Restart the Server: After making any changes to the configuration files, restart your web server to apply the changes.

For Apache:
sudo systemctl restart apache2
For Nginx:
sudo systemctl restart nginx

<?php
// Define the path to the requests file
$file = 'requests/requests.txt';

// Read the contents of the file
$requests = file_get_contents($file);

// Check if the file reading was successful
if ($requests === false) {
    echo "Error reading the requests file.";
    exit;
}

// Display the contents with HTML line breaks
echo nl2br($requests);
?>

**Comment by strawbs1989** [comment]
I'm not using a server 😕 just normal php GitHub uploader notepad++ that's
it

**Comment by strawbs1989** [comment]
And I'm using windows
